File: apps/BuddyPlay/shared/BuddyCore/Sources/BuddyCore/Models/Peer.swift
import Foundation

/// A peer is another BuddyPlay device the user can play with. The `id` is
/// generated once on first launch and persisted in `device.json`. It's not
/// a user identity — it's just a stable handle so the rivalry store can
/// keep tallies across sessions with the same person.
public struct Peer: Codable, Hashable, Sendable {
    public let id: UUID
    public var displayName: String
    public var platform: Platform
    public var lastSeenAt: Date

    public init(id: UUID, displayName: String, platform: Platform, lastSeenAt: Date) {
        self.id = id
        self.displayName = displayName
        self.platform = platform
        self.lastSeenAt = lastSeenAt
    }

    public enum Platform: String, Codable, Sendable {
        case ios
        case android
    }
}

File: apps/BuddyPlay/android/core/src/main/java/com/americangroupllc/buddyplay/core/models/Peer.kt
package com.americangroupllc.buddyplay.core.models

import kotlinx.serialization.Serializable

/**
 * A peer is another BuddyPlay device the user can play with. The [id] is
 * generated once on first launch and persisted in `device.json`. It's not
 * a user identity — just a stable handle so the rivalry store can keep
 * tallies across sessions with the same person.
 */
@Serializable
data class Peer(
    val id: String,           // UUID string
    val displayName: String,
    val platform: Platform,
    val lastSeenAt: Long,     // epoch millis
) {
    @Serializable
    enum class Platform { IOS, ANDROID }
}
